voting/ethereum/contracts/OrbsGuardians.sol

```diff
@@ -62,14 +62,18 @@ contract OrbsGuardians is IOrbsGuardians {
 
         uint i = guardiansData[msg.sender].index;
 
-        assert(guardians[i] == msg.sender);
+        assert(guardians[i] == msg.sender); // will consume all available gas.
 
+        // replace with last element and remove from end
         guardians[i] = guardians[guardians.length - 1]; // switch with last
         guardiansData[guardians[i]].index = i; // update it's lookup index
+        delete guardians[guardians.length - 1]; // remove the last one
```

Comment:
I think it is redundant to delete the last element if the array's length is decreased any how.
See further details in [this comment](https://ethereum.stackexchange.com/questions/1527/how-to-delete-an-element-at-a-certain-index-in-an-array#comment43646_1528).

Reply:
fixed in another branch